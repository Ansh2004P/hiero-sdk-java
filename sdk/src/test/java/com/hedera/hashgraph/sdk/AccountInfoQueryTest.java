package com.hedera.hashgraph.sdk;

import org.junit.AfterClass;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static io.github.jsonSnapshot.SnapshotMatcher.expect;
import static io.github.jsonSnapshot.SnapshotMatcher.start;
import static io.github.jsonSnapshot.SnapshotMatcher.validateSnapshots;

public class AccountInfoQueryTest {
    @BeforeAll
    public static void beforeAll() {
        start();
    }

    @AfterClass
    public static void afterAll() {
        validateSnapshots();
    }

    AccountInfoQuery setQuery() {
        return new AccountInfoQuery()
            .setAccountId(AccountId.fromString("0.0.5005"))
            .setMaxQueryPayment(Hbar.fromTinybar(100_000));
    }

    @Test
    @DisplayName("object to be sent matches snapshot")
    void matchesSnap() {
        expect(setQuery().makeRequest()).toMatchSnapshot();
    }
}
