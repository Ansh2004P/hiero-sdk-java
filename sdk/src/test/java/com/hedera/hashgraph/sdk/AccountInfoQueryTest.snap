com.hedera.hashgraph.sdk.AccountInfoQueryTest.matchesSnap=[
  {
    "memoizedHashCode": 0,
    "unknownFields": {
      "count": 0,
      "tags": [ ],
      "objects": [ ],
      "memoizedSerializedSize": 0,
      "isMutable": false
    },
    "memoizedSerializedSize": -1,
    "queryCase_": 9,
    "query_": {
      "memoizedHashCode": 0,
      "unknownFields": {
        "count": 0,
        "tags": [ ],
        "objects": [ ],
        "memoizedSerializedSize": 0,
        "isMutable": false
      },
      "memoizedSerializedSize": -1,
      "header_": {
        "memoizedHashCode": 0,
        "unknownFields": {
          "count": 0,
          "tags": [ ],
          "objects": [ ],
          "memoizedSerializedSize": 0,
          "isMutable": false
        },
        "memoizedSerializedSize": -1,
        "responseType_": 0,
        "initialized": true
      },
      "accountID_": {
        "memoizedHashCode": 0,
        "unknownFields": {
          "count": 0,
          "tags": [ ],
          "objects": [ ],
          "memoizedSerializedSize": 0,
          "isMutable": false
        },
        "memoizedSerializedSize": -1,
        "shardNum_": 0,
        "realmNum_": 0,
        "accountNum_": 5005,
        "initialized": true
      },
      "initialized": true
    },
    "initialized": true
  }
]